pkg/storage/events.go

```diff
@@ -30,6 +33,11 @@ type TestEventEmitterFetcher struct {
 
 // Emit emits an event using the selected storage layer
 func (e TestEventEmitter) Emit(data testevent.Data) error {
+	if e.allowedEvents != nil {
+		if _, ok := (*e.allowedEvents)[data.EventName]; !ok {
```

Comment:
does this mean that if no event is specified in the plugin, any event emission will be permitted?

Reply:
yes, this is for contest itself, it emits things like `EventTestError`, which are not specifically registered

Reply:
should we register them instead (or track this in an issue)? This feels dangerous for plugins that don't declare events but emit them

Reply:
Why? The plugin has no choice, it gets an event emitter from the pipeline that has this set and it can't change that, no way to get around. And the pipeline has its own emitter without restrictions

Reply:
(at least that is what I intended)